### PR TITLE
Context-dependent Descriptors + Electrolyte Rendering

### LINKED ISSUE #70: Create 'entry.unit_value' from units and values in the yaml file
```yaml
scan rate:
   unit: V / s
   value: 50
```

An object that prints "50 V/s" when `entry.scan_rate` is called, while unit and value are still accessible via `entry.scan_rate.value`

### PR DESCRIPTION
**Pull request by saraedum**
Fixes #70.

This PR introduces a descriptor factory that can be used to add custom features to descriptors. Here a `QuantityDescriptor` that adds a astropy `.quantity` when there are exactly `unit` and `value` as keys in the metadata.

This also implements proper rendering of such quantities and rendering of the electrolyte through the `| render` filter. Both are rendered from MarkDown templates, see `quantity.md` and `electrolyte.md`.

Finally, this fixes some bugs in the data. Namely, it replaces `M` with `mol / l` which astropy understands.

Sorry for doing several things at once in the same PR.

#### Dependencies

* [x] #92

### CODE REVIEW

File: echemdb/data/cv/descriptor.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+class GenericDescriptor:
```
Comment (saraedum) [inline echemdb/data/cv/descriptor.py:1]: I just moved this class, nothing new here.

File: echemdb/data/cv/descriptor.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+            >>> descriptor = GenericDescriptor({'a': 0})
+            >>> descriptor.yaml
+            'a: 0\n'
+
+        '''
+        import yaml
+        return yaml.dump(self._descriptor)
+
+class UnitValueDescriptor(GenericDescriptor):
```
Comment (saraedum) [inline echemdb/data/cv/descriptor.py:90]: This class is new. A descriptor wrapper for descriptors that have `value` and `unit`.

File: echemdb/data/cv/descriptor.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        from astropy import units
+
+        return float(self.value) * units.Unit(self.unit)
+
+    def __repr__(self):
+        return str(self.quantity)
+
+def Descriptor(descriptor):
+    if set(descriptor.keys()) == {"unit", "value"}:
+        return UnitValueDescriptor(descriptor)
```
Comment (saraedum) [inline echemdb/data/cv/descriptor.py:120]: We could consider just saying `return float(descriptor.value) * units.Unit(descriptor.unit)` instead.

File: echemdb/data/cv/descriptor.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    @property
+    def quantity(self):
+        from astropy import units
+
+        return float(self.value) * units.Unit(self.unit)
+
+    def __repr__(self):
+        return str(self.quantity)
+
+def Descriptor(descriptor):
```
Comment (saraedum) [inline echemdb/data/cv/descriptor.py:118]: This methods intercepts descriptor creation when a certain combination of keys is present to add more features to a descriptor.

File: templates/unit_value.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{% if quantity.unit == "K" and 250 <= quantity.value <= 400 %}
```
Comment (saraedum) [inline templates/unit_value.md:1]: I am not proposing to write temperatures in °C. This is just for demo purposes.

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -30,4 +30,4 @@ jobs:
       - name: doctest
         shell: bash -l {0}
         run: |
-          pytest --doctest-modules echemdb
+          pytest -n auto --doctest-plus --remote-data --doctest-modules echemdb
```
Comment (saraedum) [inline .github/workflows/test.yml:33]: @DunklesArchipel @linuxrider you can now use
```
pytest -n auto --doctest-plus --doctest-modules echemdb
```
to run tests more quickly, i.e., skip the tests that download the echemdb database.

File: echemdb/website/macros/render.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
+r"""
+Implements a `render` macro that renders a template with thi Jinja engine.
```
Comment (DunklesArchipel) [inline echemdb/website/macros/render.py:2]:
```suggestion
Implements a `render` macro that renders a template with the Jinja engine.
```

**Review by DunklesArchipel — APPROVED** [review]
Terrific!

File: echemdb/data/cv/descriptor.py [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+        return str(self.quantity)
+
+
+def Descriptor(descriptor):
+    r"""
+    Return `descriptor` augmented with additional convenience methods.
+
+    EXAMPLES:
+
+    Primitive types are returned unchanged:
```
Comment (saraedum) [inline echemdb/data/cv/descriptor.py:211]:
```suggestion
    Primitive types are returned unchanged::
```

**Pull request merged**